glambie/data/data_catalogue.py

```diff
@@ -152,6 +168,96 @@ def datasets_are_same_unit(self):
         unit = self.datasets[0].unit
         return all(dataset.unit == unit for dataset in self.datasets)
 
+    def average_timeseries_in_catalogue(self, remove_trend: bool = True, add_trend_after_averaging: bool = False,
+                                        out_data_group: GlambieDataGroup = GLAMBIE_DATA_GROUPS["consensus"]) \
+            -> Tuple[Timeseries, DataCatalogue]:
+        """
+        Calculates a simple average of all timeseries within the catalogue, with the option to remove trends
+        and calculate the average on the anomalies
+
+        Parameters
+        ----------
+        remove_trend : bool, optional
+            if set to True, the trend over a shared time period is removed, by default True
+        add_trend_after_averaging : bool, optional
+            this will add the average trends from all the input solutions in the catalogue to the averaged solution
+            this flag is only active when remove_trend is set to True.
+            by default False
+        out_data_group : GlambieDataGroup, optional
+            data group to be assigned to combined output Timeseries metadata,
+            by default GLAMBIE_DATA_GROUPS["consensus"]
+
+
+        Returns
+        -------
+        Tuple[Timeseries, DataCatalogue]
+            1. Timeseries object, containing the new combined solution
+            2. DataCatalogue, containing the catalogue the operation was performed on.
+               For remove_trend set to False, this is an exact copy of self.
+               For remove_trend set to True, this contains the altered datasets with the trends removed
+        """
+
+        if not self.datasets_are_same_unit():
+            warnings.warn("Warning: Trends within catalogue are not all the same unit. \
```

Comment:
If you can't convert them all to the same units, shouldn't this be an error?

Reply:
I notice there's another one or two instances of this as well. If the production of a meaningful result is not guaranteed, we should throw an error.

Reply:
raising AssertionErrors now